pkg/bgpv1/manager/reconcile.go

```diff
@@ -381,6 +385,210 @@ func (r *ExportPodCIDRReconciler) Reconcile(ctx context.Context, p ReconcilePara
+		}
+		if !found {
+			// edgecase: If the pod ip pool was removed between the call to IterKeys() and GetByKey().
+			if err := r.withdrawPool(ctx, sc, poolKey); err != nil {
+				return fmt.Errorf("failed to withdraw pod ip pool: %w", err)
+			}
+			continue
+		}
+
+		r.reconcilePodIPPool(ctx, sc, newc, pool, localPools)
+	}
+	return nil
+}
+
+// withdrawPool removes all announcements for the given pod ip pool.
+func (r *PodIPPoolReconciler) withdrawPool(ctx context.Context, sc *ServerWithConfig, key resource.Key) error {
+	advertisements := sc.PodIPPoolAnnouncements[key]
+	// Loop in reverse order so we can delete without effect to the iteration.
+	for i := len(advertisements) - 1; i >= 0; i-- {
+		advertisement := advertisements[i]
+		if err := sc.Server.WithdrawPath(ctx, types.PathRequest{Path: advertisement}); err != nil {
+			// Persist remaining advertisements
+			sc.PodIPPoolAnnouncements[key] = advertisements
+			return fmt.Errorf("failed to withdraw deleted pod ip pool route: %v: %w", advertisement.NLRI, err)
+		}
+
+		// Delete the advertisement after each withdraw in case we error half way through
+		advertisements = slices.Delete(advertisements, i, i+1)
+	}
+
+	// If all were withdrawn without error, we can delete the whole pod ip pool from the map
+	delete(sc.PodIPPoolAnnouncements, key)
+
+	return nil
+}
+
+// reconcilePodIPPool ensures the CIDRs of the given pool are announced if they are present
+// on the local node, adding missing announcements or withdrawing unwanted ones.
+func (r *PodIPPoolReconciler) reconcilePodIPPool(ctx context.Context,
+	sc *ServerWithConfig,
+	newc *v2alpha1api.CiliumBGPVirtualRouter,
+	pool *v2alpha1api.CiliumPodIPPool,
+	localPools map[string][]netip.Prefix) error {
+	poolKey := resource.NewKey(pool)
+
+	desiredRoutes, err := r.poolDesiredRoutes(newc, pool, localPools)
+	if err != nil {
+		return fmt.Errorf("poolDesiredRoutes(): %w", err)
+	}
+
+	for _, desiredRoute := range desiredRoutes {
+		// If this route has already been announced, don't add it again
+		if slices.IndexFunc(sc.PodIPPoolAnnouncements[poolKey], func(existing *types.Path) bool {
+			return desiredRoute.String() == existing.NLRI.String()
+		}) != -1 {
+			continue
+		}
+
+		// Advertise the new cidr
+		advertPathResp, err := sc.Server.AdvertisePath(ctx, types.PathRequest{
+			Path: types.NewPathForPrefix(desiredRoute),
+		})
+		if err != nil {
+			return fmt.Errorf("failed to advertise service route %v: %w", desiredRoute, err)
+		}
+		sc.PodIPPoolAnnouncements[poolKey] = append(sc.PodIPPoolAnnouncements[poolKey], advertPathResp.Path)
+	}
+
+	// Loop over announcements in reverse order so we can delete entries without effecting iteration.
+	for i := len(sc.PodIPPoolAnnouncements[poolKey]) - 1; i >= 0; i-- {
+		announcement := sc.PodIPPoolAnnouncements[poolKey][i]
+		// If the announcement is within the list of desired routes, don't remove it
+		if slices.IndexFunc(desiredRoutes, func(existing netip.Prefix) bool {
```

Comment:
Same as above regarding `slices.ContainsFunc`

Reply:
Commit b18c075 implements this change request.